osmoutils/accum/accum.go

```diff
@@ -206,6 +206,14 @@ func (accum AccumulatorObject) RemoveFromPositionCustomAcc(name string, numShare
 		return err
 	}
 
+	newNumShares := oldNumShares.Sub(numSharesToRemove)
+
+	// remove the position from state entirely if numShares = zero
+	if newNumShares.Equal(sdk.ZeroDec()) {
+		accum.DeletePosition(name)
```

Comment:
We should be deleting it in `ClaimRewards`.

If we delete it here, some uncollected rewards end up being unaccounted for

Reply:
@p0mvn  But does any position.NumShares happen in ClaimRewards?

Reply:
> Are we going to enforce calling ClaimRewards when closing positions?

That's the plan

> what do you think about leaving this here,

We cannot delete position from state here because any rewards, including unclaimed, would just disappear

Reply:
> > Are we going to enforce calling ClaimRewards when closing positions?
> 
> That's the plan
> 
> > what do you think about leaving this here,
> 
> We cannot delete position from state here because any rewards, including unclaimed, would just disappear

If this is the plan, agreed on having this check on closePosition